packages/m2m-gateway/test/integration/delegations/getProducerDelegations.test.ts

```diff
@@ -0,0 +1,152 @@
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { delegationApi, m2mGatewayApi } from "pagopa-interop-api-clients";
+import { generateId } from "pagopa-interop-models";
+import {
+  getMockedApiDelegation,
+  getMockWithMetadata,
+} from "pagopa-interop-commons-test";
+import {
+  delegationService,
+  expectApiClientGetToHaveBeenCalledWith,
+  mockInteropBeClients,
+} from "../../integrationUtils.js";
+import { PagoPAInteropBeClients } from "../../../src/clients/clientsProvider.js";
+import { getMockM2MAdminAppContext } from "../../mockUtils.js";
+import { WithMaybeMetadata } from "../../../src/clients/zodiosWithMetadataPatch.js";
+import { unexpectedDelegationKind } from "../../../src/model/errors.js";
+
+describe("getProducerDelegations", () => {
+  const mockParams: m2mGatewayApi.GetProducerDelegationsQueryParams = {
+    states: ["WAITING_FOR_APPROVAL"],
+    eserviceIds: [generateId()],
+    delegateIds: [generateId()],
+    delegatorIds: [generateId()],
+    offset: 0,
+    limit: 10,
+  };
+
+  const mockApiDelegation1 = getMockWithMetadata(
+    getMockedApiDelegation({
+      kind: delegationApi.DelegationKind.Values.DELEGATED_PRODUCER,
+    })
+  );
+  const mockApiDelegation2 = getMockWithMetadata(
+    getMockedApiDelegation({
+      kind: delegationApi.DelegationKind.Values.DELEGATED_PRODUCER,
+    })
+  );
```

Comment:
No need to have metadata here, since we add it as `undefined` to the entire response below. You can also do the same fix in `integration/delegations/getConsumerDelegations.test.ts` if it was copy pasted from there

```suggestion
  const mockApiDelegation1 =  getMockedApiDelegation({
      kind: delegationApi.DelegationKind.Values.DELEGATED_PRODUCER,
    })

  const mockApiDelegation2 = getMockedApiDelegation({
      kind: delegationApi.DelegationKind.Values.DELEGATED_PRODUCER,
    })
```